File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        stack = [root]
        
        while stack and root is not None:
            if root.left:
                stack.append(root.left)
                root = root.left
            else:
                root = stack.pop()
                root.left = None
                result.append(root.val)
                
                if root.right:
                    stack.append(root.right)
                    root = root.right
        
        return result

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    if (!root) {
        return [];
    }
    
    const answer = [];
    const stack = [root];
    
    curr = root;
    while (stack.length > 0) {        
        if (curr && curr.left) {
            stack.push(curr.left);
            curr = curr.left;
        } else {
            curr = stack.pop();
            curr.left = null;
            if (curr) {
                answer.push(curr.val);
            }
            
            if (curr && curr.right) {
                stack.push(curr.right);
                curr = curr.right;
            }
        }
    }
    
    return answer;
};